src/mappings/event/loans.ts

```diff
@@ -0,0 +1,353 @@
+            : MarketState.Pending;
+
+    // console.log(JSON.stringify(my_market));
+    await entityBuffer.pushEntity(LoanMarket.name, my_market);
+
+    console.log(`Updated ${my_market.id}`);
+}
+
+export async function activatedMarket(
+    ctx: Ctx,
+    block: SubstrateBlock,
+    item: EventItem,
+    entityBuffer: EntityBuffer
+): Promise<void> {
+    const rawEvent = new LoansActivatedMarketEvent(ctx, item.event);
+    let token = rawEvent.asV1020000;
+    const currency = currencyId.encode(token);
+    const id = currencyToString(currency);
+
+    const market = await ctx.store.get(LoanMarket, {
+        where: { id: `loanMarket_${id}` },
+    });
+    if (market === undefined) {
+        ctx.log.warn(
+            "WARNING: ActivatedMarket event did not match any existing LoanMarkets! Skipping."
+        );
+        return;
+    }
+    const height = await blockToHeight(ctx, block.height, "ActivatedMarket");
+    const activation = new LoanMarketActivation({
+        id: market.id,
+        market: market,
+        token: currency,
+        height,
+        timestamp: new Date(block.timestamp),
+    });
+    market.state = MarketState.Active
+    market.activation = activation;
+    await entityBuffer.pushEntity(LoanMarketActivation.name, activation);
+    await entityBuffer.pushEntity(LoanMarket.name, market);
+
+    console.log(`Activated ${market.id}`);
+}
+
+export async function borrow(
+    ctx: Ctx,
+    block: SubstrateBlock,
+    item: EventItem,
+    entityBuffer: EntityBuffer
+): Promise<void> {
+    const rawEvent = new LoansBorrowedEvent(ctx, item.event);
+    const [adress, currencyOfLoan, amount] = rawEvent.asV1020000;
+    const currency = currencyId.encode(currencyOfLoan);
+    const height = await blockToHeight(ctx, block.height, "LoansBorrowed");
+    const account = address.interlay.encode(adress);
+    const comment = `${getFirstAndLastFour(account)} borrowed ${await friendlyAmount(currency, Number(amount))}`;
+    await entityBuffer.pushEntity(
+        Loan.name,
+        new Loan({
+            id: item.event.id,
+            height: height,
+            timestamp: new Date(block.timestamp),
+            userParachainAddress: account,
+            token: currency,
+            amountBorrowed: amount,
+            comment: comment
+        })
+    );
+}
+
+export async function depositCollateral(
+    ctx: Ctx,
+    block: SubstrateBlock,
+    item: EventItem,
+    entityBuffer: EntityBuffer
+): Promise<void> {
+    const rawEvent = new LoansDepositCollateralEvent(ctx, item.event);
+    const [adress, depositCurr, amount] = rawEvent.asV1020000;
+    const currency = currencyId.encode(depositCurr);
+    const height = await blockToHeight(ctx, block.height, "Deposit");
+    const account = address.interlay.encode(adress);
+    let comment = '';
+    if(depositCurr.__kind==='LendToken'){
+        const newToken = await lendTokenDetails(ctx, depositCurr.value)
+        const newAmount = Number(amount) * 0.02
```

Comment:
Hardcoding the exchange rate here is incorrect. You can approximate the underlying currency value of the deposit by using the last exchange rate of the market that is stored in squid (once the `InterestAccrued` event is live).

Reply:
thank you @bvotteler for spotting this!

Reply:
Coming in phase 2.